proto/viam/app/v1/billing.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+
+package viam.app.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/v1";
+
+service BillingService {
+  rpc GetCurrentMonthUsageSummary(GetCurrentMonthUsageSummaryRequest) returns (GetCurrentMonthUsageSummaryResponse);
+  rpc GetUnpaidBalance(GetUnpaidBalanceRequest) returns (GetUnpaidBalanceResponse);
+  rpc GetInvoiceHistory(GetInvoiceHistoryRequest) returns (GetInvoiceHistoryResponse);
+  rpc GetItemizedInvoice(GetItemizedInvoiceRequest) returns (GetItemizedInvoiceResponse);
+  rpc GetBillingSummary(GetBillingSummaryRequest) returns (GetBillingSummaryResponse);
+}
+
+message GetCurrentMonthUsageSummaryRequest {
+  string org_id = 1;
+}
+
+message CurrentMonthUsageSummary {
+  double cloud_storage_usage = 1;
+  double data_upload_usage = 2;
+  double data_egres_usage = 3;
+  double standard_compute_usage = 4;
+  double total_usage_with_discount = 5; // returns number with any discounts applied
+  double total_usage_without_discount = 6; // returns number without any discounts applied
+}
+
+message GetCurrentMonthUsageSummaryResponse {
+  double cloud_storage_usage = 1;
+  double data_upload_usage = 2;
+  double data_egres_usage = 3;
+  double standard_compute_usage = 4;
+  double total_usage_with_discount = 5; // returns number with any discounts applied
+  double total_usage_without_discount = 6; // returns number without any discounts applied
+}
+
+message GetUnpaidBalanceRequest {
+  string org_id = 1;
+}
+
+message GetUnpaidBalanceResponse {
+  double unpaid_balance = 1;
+  google.protobuf.Timestamp unpaid_balance_due_date = 2;
+}
+
+message GetInvoiceHistoryRequest {
+  string org_id = 1;
+}
+
+message InvoiceSummary {
+  string id = 1;
+  google.protobuf.Timestamp invoice_date = 2;
+  double invoice_amount = 3;
+  string status = 4;
+  google.protobuf.Timestamp due_date = 5;
+}
+
+message ResourceEvent {
+  string id = 1;
+  string type = 2;
+  double usage_quantity = 3;
+  string usage_quantity_unit = 4;
+  string usage_cost = 5;
+  google.protobuf.Timestamp occurred_at = 6;
+  string user_name = 7;
+}
+
+message Invoice {
+  string id = 1;
+  google.protobuf.Timestamp invoice_date = 2;
+  double invoice_amount = 3;
+  string status = 4;
+  google.protobuf.Timestamp due_date = 5;
+  repeated ResourceEvent items = 6;
+  string emailed_to = 7;
+}
+
+message GetInvoiceHistoryResponse {
+  repeated InvoiceSummary invoices = 1;
+}
+
+message GetItemizedInvoiceRequest {
+  string id = 1;
+}
+
+message GetItemizedInvoiceResponse {
+  Invoice invoice = 1;
+}
+
+message GetBillingSummaryRequest {
+  string org_id = 1;
+}
+
+message GetBillingSummaryResponse {
+  CurrentMonthUsageSummary usage_summary = 1;
+  repeated InvoiceSummary invoices = 2;
+  double unpaid_balance = 3;
+  google.protobuf.Timestamp unpaid_balance_due_date = 4;
+  google.protobuf.Timestamp due_date = 5;
```

Comment:
what's the difference between `unpaid_balance_due_date` and `due_date`? is `due_date` for the most recent invoice and `unpaid_balance_due_date` is for the previous?

Reply:
yup correct! This needs to be a bit more scoped because we don't currently turn off services if the balance isn't paid but im matching what is needed on design with this.

Reply:
could we do `due_date` and `last_due_date`? was a bit confusing to figure out what the difference was here. also, does `unpaid_balance` include all charges up to the current moment that haven't been paid, or up to the most recent full month?

not super confident in this naming but maybe we can take inspiration from credit card statement balances?

Reply:
Agree with Emily re: modeling this after credit card statement balances.

Reply:
yeah `due_date` and `oustanding_due_date` i think that last_due_date also sounds a bit unclear?

Reply:
what if we flipped it to be `current_due_date` and `invoice_due_date`?

Reply:
okay so based on credit card statements lets do:
statement balance - all unpaid balances at the end of the last billing cycle. 
current balance - statement_balance + current month usage so far. 

Speaking to Natalia about dates

Reply:
updated except for the date -- for now, i added statement_balance_due_date + current_balance_due_date - but can change this down the line if this doesn't make sense?

Reply:
`statement_due_date` and `current_due_date`?

Reply:
`statement_balance_due_date` - when the pay the outstanding bill from last month if that is set
`current_balance_due_date` - when to pay the current bill (this should be the 15th of the Month)

The logic to actually writing these is still unclear, but im adding it in for now because its on the designs and we will want to show it.

Reply:
Okay so spoke to Natalia and we are going to remove the statement_balance_due_date. This should just be immediate i.e. the user should be prompted to just be paid immediately. 

The current_balance_due_date i.e. the day that the current balance is due is the 1st of the month i.e. when we charge if payment is set or the 15th of the current month. We may not need to pass this in from the BE, but leaving it in here for now.